### PR TITLE
Upgrade Helix to 1.0.4 in Pinot

### PR DESCRIPTION
**Pull request by jackjlli**
## Description
This PR upgrades Helix version from 0.9.8 to 1.0.4 in Pinot.

Related issue: https://github.com/apache/pinot/issues/8306

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java
Comment (qqu0127) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/ControllerTest.java:252]: Remove static instance usage

**Comment by jackjlli** [comment]
The end-to-end deployment with Helix 1.0.4 is done at LinkedIn. Here is the highlights from the new Helix release:

* All the Pinot clusters at LinkedIn are up and running with Helix 1.0.4, there is no performance downgrade after deploying Helix 1.0+.
* Zookeeper read/write access latency (P99th) dropped significantly from seconds to milliseconds, which is 1000x improvement.
* Both helix controller pipeline queued time and process time (P99th) dropped by 25-50%.
* There is no downtime during the Pinot deployment. 

Thus, we have the conclusion that the new Helix 1.0.4 release is stable, we should be good to bump up the Helix version to 1.0.4 in master branch.

**Review by mcvsubbu — APPROVED** [review]
With the kind of testing that this has gone through, I am fine merging this. 

If there are small concerns (to re-organize code, etc.) I suggest we can take it up in another PR after we merge this one.

Please give the other committers a few days in case they have any comments.

**Review by Jackie-Jiang — APPROVED** [review]
Thanks for taking time verifying the new Helix version! The effort is much appreciated!

**Pull request merged**
